sdl_android/src/main/java/com/smartdevicelink/protocol/WiProProtocol.java

```diff
@@ -172,57 +172,59 @@ else if (protocolMsg.getBulkData() != null) {
 		}
 		
 		synchronized(messageLock) {
-			if (data.length > MAX_DATA_SIZE) {
-				
-				messageID++;
-	
-				// Assemble first frame.
-				int frameCount = data.length / MAX_DATA_SIZE;
-				if (data.length % MAX_DATA_SIZE > 0) {
-					frameCount++;
-				}
-				//byte[] firstFrameData = new byte[HEADER_SIZE];
-				byte[] firstFrameData = new byte[8];
-				// First four bytes are data size.
-				System.arraycopy(BitConverter.intToByteArray(data.length), 0, firstFrameData, 0, 4);
-				// Second four bytes are frame count.
-				System.arraycopy(BitConverter.intToByteArray(frameCount), 0, firstFrameData, 4, 4);
-
-				SdlPacket firstHeader = SdlPacketFactory.createMultiSendDataFirst(sessionType, sessionID, messageID, _version,firstFrameData,protocolMsg.getPayloadProtected());
-				firstHeader.setPriorityCoefficient(1+protocolMsg.priorityCoefficient);
-				//Send the first frame
-				handlePacketToSend(firstHeader);
-				
-				int currentOffset = 0;
-				byte frameSequenceNumber = 0;
-
-				for (int i = 0; i < frameCount; i++) {
-					if (i < (frameCount - 1)) {
-	                     ++frameSequenceNumber;
-	                        if (frameSequenceNumber ==
-	                                SdlPacket.FRAME_INFO_FINAL_CONNESCUTIVE_FRAME) {
-	                            // we can't use 0x00 as frameSequenceNumber, because
-	                            // it's reserved for the last frame
-	                            ++frameSequenceNumber;
-	                        }
-					} else {
-						frameSequenceNumber = SdlPacket.FRAME_INFO_FINAL_CONNESCUTIVE_FRAME;
-					} // end-if
-					
-					int bytesToWrite = data.length - currentOffset;
-					if (bytesToWrite > MAX_DATA_SIZE) { 
-						bytesToWrite = MAX_DATA_SIZE; 
+			if (data != null) {

```

Comment:
I don't see any clarifying language that that's the intended outcome. So if that is the case, we should at least add documentation to the method that will inform users that if this method is called with a message that has no data, it will not be sent.